SpatialGDK/Source/SpatialGDK/Public/Utils/Interest/ResultTypes.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/SpatialClassInfoManager.h"
+#include "Schema/Interest.h"
+
+/**
+ * This class contains static functionality to build the required result types for query based interest.
+ *
+ * If the required result type will contain generated components, then there is a dependency on the ClassInfoManager to retrieve those.
+ * Otherwise the result types are built using the predefined component sets in SpatialConstants.
+ */
+
+namespace SpatialGDK
+{
+
+class SPATIALGDK_API ResultTypes
```

Comment:
Nit: feel like I would expect this to be an object instead of a class full of static functions. Maybe have it be called `ResultTypeFactory`, `ResultTypeHelpers`, etc.?

Reply:
As in, keep it static but rename it to `ResultTypeHelpers` to make it clearer what it does? Or change it to an object initialised with the class info manager?

Reply:
Ah yeah my comment was a bit confusing. Keeping the class full of static functions is fine, just renaming it to be clearer would be better.

Reply:
Would like to avoid `helpers` or `utils` though, and I don't think it's a factory- would @aleximprobable's suggestion of making this `InterestQueryResultTypes` to make the contents a little more explicit work?

Reply:
I don't think that name helps here because it only qualifies what kind of ResultType it is: an InterestQuery one. It doesn't help qualify what this class does.

Reply:
Gone with `ResultTypeCreators` as I agree it didn't get across what the class was for before. I think this is clear now, let me know if it is still misleading in some way.

Reply:
I like the idea, that the naming explains what the role is. However, it doesn't seem obvious to me, what the Result Type is itself. I could accept it though, since this functionality is internal to the `SpatialClassInfoManager`.

Reply:
Why not just make them free functions?

Reply:
Alternatively, static functions that take a single pointer as an argument is the C style way of making member functions on the argument class type.

Reply:
made them free functions.